classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -225,10 +231,9 @@ function execute($args, $request) {
 			}
 			// Add or update comments to editor
 			if ($this->getData('commentsToEditor')){
-				$queries = $queryDao->getByAssoc(ASSOC_TYPE_SUBMISSION, $this->submissionId);
-				$query = $queries->next();
+				$query = $this->getCommentsToEditor($this->submissionId);				
 				if (isset($query)){
```

Comment:
I believe you do not need this `if-else` -- you can just call the `setCommentsToEditor` with the `$query`, because the `$query` is then checked there.